packages/forms-web-app/src/controllers/final-comment/comments-question.js

```diff
@@ -0,0 +1,38 @@
+const { VIEW } = require('../../lib/views');
+
+exports.getCommentsQuestion = async (req, res) => {
+	let hasComments = null;
+
+	if (
+		req.session &&
+		req.session.finalComment &&
+		Object.prototype.hasOwnProperty.call(req.session.finalComment, 'hasComments')
+	) {
+		hasComments = req.session.finalComment.hasComments;
+	}
+
+	res.render(VIEW.FINAL_COMMENT.COMMENTS_QUESTION, {
+		hasComments
+	});
+};
+
+exports.postCommentsQuestion = async (req, res) => {
+	const {
+		body,
+		body: { errors = {}, errorSummary = [] }
+	} = req;
+
+	if (Object.keys(errors).length > 0) {
+		return res.render(VIEW.FINAL_COMMENT.COMMENTS_QUESTION, {
+			errors,
+			errorSummary
+		});
+	}
+
+	const hasComments = body['comments-question'] === 'yes';
+	req.session.finalComment = { hasComments: hasComments };
```

Comment:
This should probably account for the possibility that the finalComment object could already exist (for example if the user has navigated back from a subsequent step)

Reply:
i've put in the 'dev only' workaround from your earlier ticket